lib/drip/client/configuration.rb

```diff
@@ -0,0 +1,21 @@
+module Drip
+  class Client
+    class Configuration
+      DEFAULT_URL_PREFIX = "https://api.getdrip.com/".freeze
+      private_constant :DEFAULT_URL_PREFIX
+
+      CONFIGURATION_FIELDS = %i[access_token api_key account_id url_prefix http_open_timeout http_timeout].freeze
+
+      attr_accessor(*CONFIGURATION_FIELDS)
+
+      def initialize(access_token: nil, api_key: nil, account_id: nil, url_prefix: DEFAULT_URL_PREFIX, http_open_timeout: nil, http_timeout: nil) # rubocop:disable Metrics/ParameterLists
```

Comment:
Can we use keyword arguments splat here, e.g:

```ruby
def initialize(**options)
  options.each do |k, v|
    public_send("#{k}=", v)
  end
end
```

It will raise `NoMethodError` when unknown option is given but you can raise custom error if you want.

Reply:
Good idea! Added.